### PR TITLE
New version: Google.CloudSDK version 477.0.0

### PR DESCRIPTION
**Pull request by aisgbnok**
Checklist for Pull Requests
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/154823)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335146

(Automated response - build 881.)

**Comment by stephengillie** [comment]
`exit code -1978335146` is new to me. Researching it suggests it might mean "The installer cannot be run from an administrator context" - does this package require `ElevationRequirement: elevationProhibited`?

**Comment by aisgbnok** [comment]
Hmm, that's odd.

In my testing running with `/S /allusers` the installer will elevate itself for machine scope installation and succeed. user scope installation it does require non-elevated context, but I believe I have the metadata set accordingly.

https://github.com/microsoft/winget-pkgs/blob/31e41e72102987dfc9f3c07e61807cd14d3f5093/manifests/g/Google/CloudSDK/477.0.0/Google.CloudSDK.installer.yaml#L20-L41

I'll do some more manual validation tonight and get back to you.

**Comment by aisgbnok** [comment]
@stephengillie,

Apologies for the late response, but I finally had a chance to retest. Unfortunately, I'm still not sure what's wrong.

I have the the installer with `Scope: user` set to `elevationProhibited`, and `Scope: machine` with `elevatesSelf`.

### `Scope: user`

#### Administrator Shell

image
image

When testing from an Administrator shell WinGet gracefully fails. This is not an error from the package installer, but WinGet. Tested locally and in Sandbox.

#### User Shell

image

When testing from a user shell WinGet successfully installs the package to the user's app data.

### `Scope: machine`

#### Administrator Shell

image
image


When testing from an Administrator shell WinGet successfully installs the package to the machine / program files. Tested locally and in Sandbox.

#### User Shell

image

When testing from a user shell WinGet successfully elevates the installer through UAC. The package is successfully installed to the machine / program files.

### TLDR

I tested all four variations of installs (Scope Machine/User x Shell Admin/User) and they all installed or failed gracefully as expected.

**Comment by aisgbnok** [comment]
After the testing above these are my rough thoughts now.

### Windows Version?

Maybe it's the windows build version. I'm running and doing all my testing on Windows 11 10.0.22635.3646. I have tested both WinGet Stable 1.7 and WinGet 1.8 Preview, so that's not it.

The pipeline and your manual validation seems to be on Windows 10.

I'm away from my main dev PC atm so I don't have access to a Windows 10 VM to test.

### x64?

Maybe the interactive installer only installs x64, and Google simply doesn't list on their page: https://cloud.google.com/sdk/docs/downloads-interactive

[Though they have x86 builds.](https://cloud.google.com/sdk/docs/downloads-versioned-archives#installation_instructions)

Additionally, previous PRs haven't had this issue. #136092
Maybe something has changed w/ vanity installer, or has WinGet/validation pipeline updated?

**Comment by aisgbnok** [comment]
I'm going to rebase and rerun the pipeline for now because I'm curious.

**Comment by aisgbnok** [comment]
@stephengillie,

Take a look at this PR, specifically comment: https://github.com/microsoft/winget-pkgs/pull/155308#issuecomment-2133957238

---

There are only ~4 packages in winget-pkgs that use `ElevationRequirement: elevationProhibited`, and only ~1 that uses it in the individual installer nodes vs root. [It's not recent so it might not have had this issue.](https://github.com/microsoft/winget-pkgs/blob/3c01c3f77496188653012ac3c3f5b40ed17724c0/manifests/m/Microsoft/VisualStudioCode/1.67.2/Microsoft.VisualStudioCode.installer.yaml#L19)

So I took Spotify.Spotify which has `elevationProhibited` in the root. And I was able to get it to validate in the root, but by simply moving to individual nodes it fails validation. That's this: https://github.com/microsoft/winget-pkgs/pull/155308#issuecomment-2133957238

I'm unsure how to proceed.

**Comment by aisgbnok** [comment]
Okay, so I found this comment: https://github.com/microsoft/winget-pkgs/pull/124730#pullrequestreview-1718023613

That made me realize this manifest could be refined to use that approach. Only because the CloudSDK installer does have a `/singleuser` switch that it thankfully respects even in admin shell. I tested locally and it seems to be working. And everything is passing in the pipeline now.

However, I am still curious as to why https://github.com/microsoft/winget-pkgs/pull/154823#issuecomment-2133964110 is the case?

**Review by stephengillie — APPROVED** [review]
